File: LV1/1.7/jewelrybox/jewelrybox.cpp
#include <cmath>
#include <cstdio>

double find_height(double a, double b) {
    int ab = a + b;
    return (ab - sqrt(ab * ab - 3 * a * b)) / 6.0;
}

double max_volumn(double a, double b) {
    double m_h = find_height(a, b);
    return m_h * (a - 2 * m_h) * (b - 2 * m_h);
}

int main() {
    int n;
    double a, b;
    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        scanf("%lf %lf", &a, &b);
        printf("%lf\n", max_volumn(a, b));
    }
}

File: LV1/1.7/jewelrybox/jewelrybox.go
package main

import (
	"fmt"
	"math"
)

func findHeight(a float64, b float64) float64 {
	ab := a + b
	return (ab - math.Sqrt(ab*ab-3*a*b)) / 6.0
}

func maxVolumn(a float64, b float64) float64 {
	mH := findHeight(a, b)
	return mH * (a - 2*mH) * (b - 2*mH)
}

func main() {
	var n int
	var a, b float64
	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%f %f", &a, &b)
		fmt.Println(maxVolumn(a, b))
	}
}
